pkg/validation/spec/gnostic.go

```diff
@@ -0,0 +1,1226 @@
+		if err := k.Paths.FromGnostic(g.Paths); err != nil {
+			return err
+		}
+	}
+
+	if g.Definitions != nil {
+		k.Definitions = make(Definitions, len(g.Definitions.AdditionalProperties))
+		for _, v := range g.Definitions.AdditionalProperties {
+			converted := Schema{}
+			if err := converted.FromGnostic(v.Value); err != nil {
+				return err
+			}
+			k.Definitions[v.Name] = converted
+
+		}
+	}
+
+	if g.Parameters != nil {
+		k.Parameters = make(
+			map[string]Parameter,
+			len(g.Parameters.AdditionalProperties))
+		for _, v := range g.Parameters.AdditionalProperties {
+			p := Parameter{}
+			if err := p.FromGnostic(v.Value); err != nil {
+				return err
+			}
+
+			k.Parameters[v.Name] = p
+		}
+	}
+
+	if g.Responses != nil {
+		k.Responses = make(
+			map[string]Response,
+			len(g.Responses.AdditionalProperties))
+
+		for _, v := range g.Responses.AdditionalProperties {
+			p := Response{}
+			if err := p.FromGnostic(v.Value); err != nil {
+				return err
+			}
+
+			k.Responses[v.Name] = p
+		}
+	}
+
+	if g.SecurityDefinitions != nil {
+		k.SecurityDefinitions = make(SecurityDefinitions)
+		if err := k.SecurityDefinitions.FromGnostic(g.SecurityDefinitions); err != nil {
+			return err
+		}
+	}
+
+	if len(g.Security) > 0 {
+		k.Security = make([]map[string][]string, len(g.Security))
+		for i, v := range g.Security {
+			if len(v.AdditionalProperties) == 0 {
+				continue
+			}
+			k.Security[i] = make(map[string][]string, len(v.AdditionalProperties))
+
+			converted := k.Security[i]
+			for _, p := range v.AdditionalProperties {
+				if p.Value != nil {
+					if len(p.Value.Value) > 0 {
+						converted[p.Name] = p.Value.Value
+					} else {
+						converted[p.Name] = nil
+					}
+				} else {
+					converted[p.Name] = nil
+				}
+			}
+		}
+	}
+
+	if len(g.Tags) > 0 {
+		k.Tags = make([]Tag, len(g.Tags))
+		for i, v := range g.Tags {
+			if err := k.Tags[i].FromGnostic(v); err != nil {
+				return err
+			}
+		}
+	}
+
+	if g.ExternalDocs != nil {
+		k.ExternalDocs = &ExternalDocumentation{}
+		if err := k.ExternalDocs.FromGnostic(g.ExternalDocs); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Info
+
+func (k *Info) FromGnostic(g *openapi_v2.Info) error {
+	if g == nil {
+		return nil
+	}
+
+	if err := k.VendorExtensible.FromGnostic(g.VendorExtension); err != nil {
+		return err
+	}
+	return k.InfoProps.FromGnostic(g)
+}
+
+func (k *InfoProps) FromGnostic(g *openapi_v2.Info) error {
+	if g == nil {
+		return nil
+	}
+
+	k.Description = g.Description
+	k.Title = g.Title
+	k.TermsOfService = g.TermsOfService
+
+	if g.Contact != nil {
+		k.Contact = &ContactInfo{}
+		k.Contact.FromGnostic(g.Contact)
+	}
+
+	if g.License != nil {
+		k.License = &License{}
+		k.License.FromGnostic(g.License)
+	}
+
+	k.Version = g.Version
+	return nil
+}
+
+func (k *License) FromGnostic(g *openapi_v2.License) error {
+	if g == nil {
+		return nil
+	}
+
+	// License does not embed to VendorExtensible!
+	// data loss from g.VendorExtension
```

Comment:
we know there are field mismatches, and we don't expect this mismatched fields to occur in kube openapi we're parsing, but it seems like we should propagate the fact that there was data loss up so it's visible at the top (we can check in tests, and consumers would at least have visibility to warn or handle as they see fit). That would change return signatures to `(ok bool, err error)`, and we could flip `ok` false for data loss scenarios

Reply:
i've changed the signature to now return an `ok` result